data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java

```diff
@@ -146,43 +141,42 @@ private static CircuitBreakerOptions createCircuitBreakerOptions(final DataPlane
     return new CircuitBreakerOptions();
   }
 
-  protected io.vertx.kafka.client.producer.KafkaProducer<String, CloudEvent> createProducer(
+  protected KafkaProducer<String, CloudEvent> createProducer(
     final Vertx vertx,
     final DataPlaneContract.Resource resource,
     final DataPlaneContract.Egress egress) {
 
     // producerConfigs is a shared object and it acts as a prototype for each consumer instance.
-    final var producerConfigs = (Properties) this.producerConfigs.clone();
-    producerConfigs.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, resource.getBootstrapServers());
+    final var producerConfigs = this.producerConfigs.entrySet()
+      .stream()
+      .map(e -> new SimpleImmutableEntry<>(e.getKey().toString(), e.getValue().toString()))
+      .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
 
-    final var kafkaProducer = new KafkaProducer<>(
-      producerConfigs,
-      new StringSerializer(),
-      new CloudEventSerializer()
-    );
+    // TODO create a single producer per bootstrap servers.
```

Comment:
?!

Reply:
It's unrelated but we can do the same optimization of the other PR, having one shared producer per bootstrap server that handles responses.

Reply:
Aren't we already reusing the same producers? If not, then that's fine let's keep it

Reply:
We did it on the receiver side, on the dispatcher side we haven't.